_pages/a-pairing-session-template.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: default
+title: A Pairing Session Template
+permalink: /template
+meta: A template to help guide your pair programming session
+---
+
+{% include article-top.html %}
+
+Here's a template for your next pairing session:
+
+```
+1. [ ] Agree on the high-level goal out loud.
+2. [ ] Break the work into a handful of tasks.
+3. [ ] Come up with a preliminary ordering for those tasks.
+4. [ ] Decide your driver/navigator swapping strategy.
+5. [ ] Configure git to share credit.
+6. [ ] Eliminate distractions.
+7. [ ] Work.
+8. [ ] Analyze the session.
+```
+
+Let's talk more about each step.
+
+
+## 1. Agree on the high-level goal
+
+State **out loud** what you hope to accomplish at a high level.
+
+You wouldn't think it'd be possible for two people to start pairing without agreement about where they're headed, but it's surprisingly easy.
+
+
+## 2. Break the goal into smaller tasks and order them
+
+It's worth trying to break your high-level goal into a handful of smaller steps.
+
+This has a number of benefits:
+
+- It makes the goal less intimidating.
+- You'll spot dead ends and pitfalls more easily.
+- You can sort your task list by priority.
+- You're more likely to notice that accomplishing task C would make B easier, and reorder appropriately.
+- You can decide on a task based on your current energy levels.
+
+[Some folks](https://www.jamesshore.com/Agile-Book/pair_programming.html) like
+to write each task on its own index card. The stack of them lives in front of the navigator.
+Each card can be a nice home for notes or ideas to bring up when there is a break in
+the action.
+
+
+## 3. Decide what will trigger a driver/navigator swap
+
+Unless you already know what works best for you, I strongly recommend the Pomodoro Technique:
+
+1. Code for 25 minutes.
+2. Take a 5 minute break.
+3. Switch drivers.
+
+Other [pair programming styles](/pair-programming-guide-styles) exist if you wish to try them.
+
+
+## 4. Configure git to share credit
+
+If two of you work on some code, both your names should appear on the commit.
+
+Here's [a handy guide](https://help.github.com/articles/creating-a-commit-with-multiple-authors/) to configuring git appropriately.
```

Comment:
Could be worth calling out some of the other pairing commit options out there?  Some I've used include [git pair](https://github.com/chrisk/git-pair), [git duet](https://github.com/git-duet/git-duet), and [git-together](https://github.com/kejadlen/git-together)